perf/tools/setup_env.sh

```diff
@@ -0,0 +1,134 @@
+#!/bin/sh
+
+# The script setting up a Linux operating system before running
+# Tarantool benchmarks.
+# See https://github.com/tarantool/tarantool/wiki/Benchmarking.
+
+set -eu
+
+uid=$(id -u)
+if [ "$uid" -ne 0 ]
+  then echo "Please run as root."
+  exit 1
+fi
+
+###
+# Helpers.
+###
+
+cpu_vendor="unknown"
+cpuinfo_vendor=$(awk '/vendor_id/{ print $3; exit }' < /proc/cpuinfo)
+if [ "$cpuinfo_vendor" = "GenuineIntel" ]; then
+   cpu_vendor="intel"
+elif [ "$cpuinfo_vendor" = "AuthenticAMD" ]; then
+   cpu_vendor="amd"
+else
+   echo "Unknown CPU vendor '$cpuinfo_vendor'"
+   exit 1
+fi
+
+###
+# Kernel command line parameters.
+###
+
+echo -n "Disable AMD SMT or Intel Hyperthreading "
+sysfs_path="/sys/devices/system/cpu/smt/active"
+if [ -f "$sysfs_path" ]; then
+  is_set=$(cat $sysfs_path)
+  [ "$is_set" = 1 ] && echo "SUCCESS" || echo "FAILED (hint: set 'nosmt' kernel parameter)"
+else
+  echo "SKIPPED"
+fi
+
+echo -n "Isolate CPUs for benchmarking "
+sysfs_path="/sys/devices/system/cpu/isolated"
+if [ -f "$sysfs_path" ]; then
+  isolcpus=$(cat "$sysfs_path")
+  [ -n "$isolcpus" ] && echo "SUCCESS" || echo "FAILED (hint: set 'isolcpus' kernel parameter)"
```

Comment:
Minor: Since fail doesn't mean stop of the script, I suggest renaming `FAILED` to the `WARNING` or something like this. Similary, we may rename `SUCCESS` -> `CHECKED`. Here and below for checking routines.
Feel free to ignore.

Reply:
I like the idea, thanks.
Introduced variables with user messages:
```diff
--- a/perf/tools/setup_env.sh
+++ b/perf/tools/setup_env.sh
@@ -27,6 +27,10 @@ else
    exit 1
 fi
 
+FAILURE_MSG="WARNING"
+SUCCESS_MSG="CHECKED"
+SKIPPED_MSG="SKIPPED"
+
 ###
 # Kernel command line parameters.
 ###
```
and these variables used in all messages in a whole script.

Reply:
Nice, thanks!